main.go

```diff
@@ -42,31 +43,51 @@ func SyncPluginXML(path string) error {
 }
 
 // AddLicenceTo ...
-func AddLicenceTo(path string, extension string, licence string,startTag string, endTag string) error {
+func AddLicenceTo(path string, extension string, licence string) error {
 	if len(licence) == 0 {
 		licence = "writer/licence"
 	}
+	if path == "" {
+		path = "."
+	}
 	files, err := reader.FilePathWalkDir(path)
 	if err != nil {
 		return err
 	}
-
-	for _, p := range files {
-		ext := filepath.Ext(p)
-		if ext == extension || len(extension) == 0 {
-			writer.WriteLicenceToFile(p, licence,startTag,endTag)
+	extensions := make(map[string][]string)
+	extensions[".html"] = []string{"<!--", "-->"}
+	extensions[".java"] = []string{"/*", "*/"}
+	extensions[".js"] = []string{"/*", "*/"}
+	extensions[".ts"] = []string{"/*", "*/"}
+	extensions[".py"] = []string{"\"\"\"", "\"\"\""}
+	if extension == "" {
```

Comment:
if and else blocks are nearly the same code. You could solve this only by adding another statement to if like
```golang
if ext == extension || extension == "" {
  // this if statement would allow you to licence only given extension
  // or licence every file if no extension is given.
}
```
Or you can extract this for loop to a function and make it better. But the main idea is, this code needs an improvement.,

Reply:
Fixed it.